packages/qdrant-loader/tests/unit/core/worker/test_job_handlers.py

```
@@ -365,3 +366,130 @@ async def test_incremental_pull_wraps_orchestrator_exception_as_transient(monkey
+    class MockGraphStore:
+        async def export_graph(self):
+            return (
+                [{"id": "A"}, {"id": "B"}],
+                [{"source": "A", "target": "B"}],
+            )
+
+        async def update_clusters_batch(self, updates):
+            self.updates = updates
+
+    # fake louvain
+    def fake_partition(graph, random_state=None):
```

Ruff (0.15.14)

[warning] Missing return type annotation for private function `fake_partition`

(ANN202)